modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/SkuResourceImpl.java

```diff
@@ -245,9 +250,22 @@ public Sku postProductIdSku(Long id, Sku sku) throws Exception {
 	private Sku _addOrUpdateSKU(CPDefinition cpDefinition, Sku sku)
 		throws Exception {
 
+		ServiceContext serviceContext = _serviceContextHelper.getServiceContext(
+			cpDefinition.getGroupId());
+
 		CPInstance cpInstance = SkuUtil.addOrUpdateCPInstance(
-			_cpInstanceService, sku, cpDefinition,
-			_serviceContextHelper.getServiceContext(cpDefinition.getGroupId()));
+			_cpInstanceService, sku, cpDefinition, serviceContext);
+
+		if (Objects.equals(
+				SkuUtil.getCommercePricingConfigurationKey(
```

Comment:
I think you should add the same logic to the addOrUpdate method of the Util class